atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -84,18 +84,33 @@ public HikariCPConnectionManager(ConnectionConfig connConfig) {
 
     @Override
     public Connection getConnection() throws SQLException {
-        long start = System.currentTimeMillis();
-        try {
-            return getConnectionInternal();
-        } finally {
-            long now = System.currentTimeMillis();
-            long elapsed = now - start;
-            if (elapsed > 1000) {
-                log.warn("Waited {}ms for connection", elapsed);
-                logPoolStats();
-            } else {
-                log.debug("Waited {}ms for connection", elapsed);
+        while (true) {
+            long start = System.currentTimeMillis();
```

Comment:
you probably want this before the loop starts as we really want to warn if time to acquire connection from user perspective is over the threshold